### DISCUSSION TITLE
No apparent check about expired Green Pass

### DISCUSSION THREAD

**Discussion by saralori** [masterix21/green-pass#8] [General]
Hi, I am currently using this library and just today I have noticed that there not seems to be any check to verify the validity of the Green Pass.

Explaining better: I have used the QR Code of an expired Green Pass, and I receive the relative information but any advise about the fact that it is expired.

Is it normal?

Thanks.

**Comment by masterix21** [comment]
Hi @saralori, yes, it's normal. 

The package offers a way to decode the greenpass but doesn't implement any validation rules due to each country's complexity. I was working on the extended version to include the validation rules, but the work is on stand-by because the free time isn't enough, the package is free-for-use, and I didn't find a sponsor willing to support my efforts.

If you are interested in it, you could help me and the community implement the validation rules for free.

**Comment by saralori** [comment]
Hi @masterix21 , thanks for the quick response.

I would like to help you, if I will be able to do it. Do you have any indication to share with me to implement the validation rules? For instance free APIs available, etc...

Thank you very much for the support and for the work done up to now.
